newroutes/settings.js

```diff
@@ -0,0 +1,189 @@
+
+// Import middleware
+const {
+  attachReadRouteHandlerWrapper,
+  attachRollbackRouteHandlerWrapper,
+} = require("@middleware/routeHandler")
+
+const { UpdateSettingsRequestSchema } = require("@validators/RequestSchema")
+
+const extractConfigFields = (config) => ({
+  url: config.content.url,
+  title: config.content.title,
+  favicon: config.content.favicon,
+  shareicon: config.content.shareicon,
+  is_government: config.content.is_government,
+  facebook_pixel: config.content["facebook-pixel"],
+  google_analytics: config.content.google_analytics,
+  linkedin_insights: config.content["linkedin-insights"],
+  resources_name: config.content.resources_name,
+  colors: config.content.colors,
+})
+const extractFooterFields = (footer) => footer.content
+const extractNavFields = (navigation) => ({
+  logo: navigation.content.logo,
+})
+
+const mergeUpdatedData = (currentData, updatedData) => {
+  const clonedCurrentData = _.cloneDeep(currentData)
+  Object.keys(updatedData).forEach((field) => {
+    clonedCurrentData[field] = updatedData[field]
+  })
+  return clonedCurrentData
+}
+
+class SettingsRouter {
+  constructor({
+    configYmlService,
+    footerYmlService,
+    navYmlService,
+    homepagePageService,
+  }) {
+    this.configYmlService = configYmlService
+    this.footerYmlService = footerYmlService
+    this.navYmlService = navYmlService
+    this.homepagePageService = homepagePageService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  async readSettingsPage(req, res) {
+    const { accessToken } = req
+    const { siteName } = req.params
+    const reqDetails = { siteName, accessToken }
+
+    const { config, footer, navigation } = await this.retrieveSettingsFiles(
+      reqDetails
+    )
+
+    return res.status(200).json({
+      configSettings: extractConfigFields(config),
+      footerSettings: extractFooterFields(footer),
+      navigationSettings: extractNavFields(navigation),
+    })
+  }
+
+  async updateSettingsPage(req, res) {
+    const { accessToken, body } = req
+    const { siteName } = req.params
+
+    const { error } = UpdateSettingsRequestSchema.validate(req.body)
+    if (error) throw new BadRequestError(error)
+
+    const reqDetails = { siteName, accessToken }
+
+    const {
+      config,
+      footer,
+      navigation,
+      homepage,
+    } = await this.retrieveSettingsFiles(reqDetails, true)
+
+    // extract data
+    const {
+      configSettings: updatedConfigContent,
+      footerSettings: updatedFooterContent,
+      navigationSettings: updatedNavigationContent,
+    } = body
+
+    if (!_.isEmpty(updatedConfigContent)) {
+      const mergedConfigContent = mergeUpdatedData(
+        config.content,
+        updatedConfigContent
+      )
+      await this.configYmlService.update(reqDetails, {
+        fileContent: mergedConfigContent,
+        sha: config.sha,
+      })
+
+      // update homepage title only if necessary
```

Comment:
We actually should be updating the description in the homepage as well - see https://github.com/isomerpages/isomercms-backend/pull/280 for the changes (has not been merged yet)

Reply:
Fixed with [9e1e3b8](https://github.com/isomerpages/isomercms-backend/pull/282/commits/9e1e3b8464be1275dce87c557b4e2ab194152f59)